elks/arch/i86/kernel/irq.c

```diff
@@ -132,24 +140,23 @@ void INITPROC irq_init(void)
 {
     flag_t flags;
 
-    init_irq();
+    init_irq();  // PIC initialization
+
+	// TODO: no more need to disable the timer
+	// BEFORE the interrupt initialization
+	// as timer vector is kept as is
 
+	// TODO: move that timer stuff in `timer.c`
+	// to clearly separate the PIC and PIT driving
     disable_timer_tick();
 
-    /* Old IRQ 8 handler is nuked in this routine */
-    irqtab_init();			/* Store DS */
+    irqtab_init();  // now only saves previous vector 08h (timer)
+	int_handler_add (0x80, 0x80, _irqit);  // system call
 
     /* Set off the initial timer interrupt handler */
-    if (request_irq(TIMER_IRQ, timer_tick, NULL))
-	panic("Unable to get timer");
+    if (request_irq(TIMER_IRQ, timer_tick, INT_GENERIC))
+    	panic("Unable to get timer");
 
-    /* Re-start the timer only after irq is set */
+    /* Re-start the timer */
     enable_timer_tick();
-
-    if (sys_caps & CAP_IRQ2MAP9) {	/* PC/AT or greater */
-	save_flags(flags);
-	clr_irq();
-	enable_irq(2);		/* Cascade slave PIC */
-	restore_flags(flags);
-    }
```

Comment:
For reasons previously stated, I would like to revert this change for this PR. Changing this has possible bad effect base on unknown legacy BIOS behavior.

If we stick all the PC BIOS legacy problem behavior in a seperate PR, it will be easy to rollback for systems that exhibit problems when finally tested, which won't be before this PR committed.

Reply:
Again, I cannot figure out how AT BIOS services (HDD...) could work without enabling the 8259 cascade...

Reply:
How about we reach a compromise here, and wrap the code for the enable_irq's with \#if 0 for now?

That would allow me to fix things in the event of a problem quickly, rather than muddling through huge changes in this PR.

After testing, the code could be removed. I don't have time for complete regression testing now, but other users could quickly test on their hardware, and easily change if 0 to if 1 etc for remote debugging help.